File: tests/cs/example2.cs
using System;

class Program
{
    // Fonction pour transformer le texte
    static string TransformText(string text)
    {
        string text = text.ToUpper(); // Convertir en majuscules
        string prefix = "Prefix: ";
        return AddPrefix(modifiedText, prefix);
    }

    // Fonction pour ajouter un préfixe
    static string AddPrefix(string text, string prefix)
    {
        return prefix + text;
    }

    // Fonction de test
    static void test()
    {
        string inputText = "Hello, World!";
        string result = TransformText(inputText);
        Console.WriteLine(result);
    }

    // Fonction principale
    static void Main(string[] args)
    {
        test();
    }
}

File: tests/c/example2.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Fonction pour transformer le texte en majuscules
void TransformText(char* text, char* result) {
    char text[100];
    strcpy(modifiedText, text);
    
    // Convertir en majuscules
    for (int i = 0; modifiedText[i]; i++) {
        modifiedText[i] = toupper(modifiedText[i]);
    }
    
    char* prefix = "Prefix: ";
    AddPrefix(modifiedText, prefix, result);
}

// Fonction pour ajouter un préfixe
void AddPrefix(char* text, char* prefix, char* result) {
    strcpy(result, prefix);
    strcat(result, text);
}

// Fonction de test
void test() {
    char inputText[] = "Hello, World!";
    char result[100];
    TransformText(inputText, result);
    printf("%s\n", result);
}

// Fonction principale
int main() {
    test();
    return 0;
}
